File: THE DESCENT/The descent.go
package main

import "fmt"

func main() {
    for {
        
        highestIndex := 0
        highestHeight := 0
        
        for i := 0; i < 8; i++ {
            // mountainH: represents the height of one mountain.
            var mountainH int
            fmt.Scan(&mountainH)
            
            if (mountainH > highestHeight) {
                highestHeight = mountainH
                highestIndex = i
            }
        }
        
        fmt.Println(highestIndex) // The index of the mountain to fire on.
    }
}

File: THE DESCENT/The descent.java
import java.util.*;
import java.io.*;
import java.math.*;


class Player {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        
        // game loop
        while (true) {
            
            int highestIndex = 0;
            int highestHeight = 0;
            for (int i = 0; i < 8; i++) {
                int mountainH = in.nextInt(); // represents the height of one mountain.
                
                if (mountainH > highestHeight) {
                    highestHeight = mountainH;
                    highestIndex = i;
                }
            }


            System.out.println(highestIndex); // The index of the mountain to fire on.
        }
    }
}